### ISSUE TITLE
Extend support to all POSIX file types where possible

### ISSUE THREAD

**Issue by adammoody** [hpc/mpifileutils#315]
The suite currently supports directories (S_IFDIR), regular files (S_IFREG), and symlinks (S_IFLNK) uniformly across all tools and common library API calls.  We should expand the suite to handle all POSIX types by adding sockets (S_IFSOCK), block devices (S_IFBLK), character devices (S_IFCHR), and FIFOs (S_IFIFO) where possible.  For tools/calls where that does not make sense or more time is needed to add support, we can document the limitations, check for those types while processing, and print warnings/errors.

**Comment by mkgilbert** [comment]
+1 for adding support for socket types. I just ran into an issue today where I couldn't verify a dcp because a user had a socket file buried in their data. I'm not sure how generally helpful it is to be able to make copies of socket files, but at the very least, having the feature would make verification of copies more stable.

**Comment by akesandgren** [comment]
Yes +1 on this. It is not common but fifos and sockets are common enough to warrant handling them correctly.

Shouldn't be that hard to fix...
